filepicker/filepicker.go

```diff
@@ -7,25 +7,20 @@ import (
 	"sort"
 	"strconv"
 	"strings"
-	"sync"
+	"sync/atomic"
 
 	"github.com/charmbracelet/bubbles/key"
 	tea "github.com/charmbracelet/bubbletea"
 	"github.com/charmbracelet/lipgloss"
 	"github.com/dustin/go-humanize"
 )
 
-var (
-	lastID int
-	idMtx  sync.Mutex
-)
+var lastID atomic.Int64
 
 // Return the next ID we should use on the Model.
 func nextID() int {
-	idMtx.Lock()
-	defer idMtx.Unlock()
-	lastID++
-	return lastID
+	lastID.Add(1)
+	return int(lastID.Load())
```

Comment:
add returns the new value, so this can be:

```suggestion
	return int(lastID.Add(1))
```